### PR TITLE
CMM-844/CMM-842: help center and logs in the new support screen

### PR DESCRIPTION
**Pull request by adalpari**
## Description

This PR is closing CMM-844 and CMM-842:

- Linking the "Help Center" entry to the Help Center web page
- Creating a new "Application Logs" screen. The screen is showing the current stored logs clasificated by day, and allowing the user to share them.
- Adding the app version to the Support screen

## Testing instructions

1. Log into the app with a self-hosted site
2. Enable MODERN_SUPPORT Experimental Feature flag
3. Go to "Me screen" -> Help & Support
- [x] Verify "Help Center" opens the Help Center web page
- [x] Verify "Application Logs" shows the new Application Logs screen (Logs lists and logs details)
- [x] Verify share logs works

1. Log into WP.COM
2. Enable MODERN_SUPPORT Experimental Feature flag
3. Go to "Me screen" -> Help & Support
- [x] Verify "Help Center" opens the Help Center web page
- [x] Verify "Application Logs" shows the new Application Logs screen (see the video)
[ ] Verify share logs works

https://github.com/user-attachments/assets/b14bf0a3-24e8-4e0e-97bd-fbf024ba8ccd

https://github.com/user-attachments/assets/16779561-5182-4830-8819-a1a7ff7a4ad7

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by oguzkocer — CHANGES_REQUESTED** [review]
Testing worked as expected. As part of my PR review flow, I always ask Claude to review it alongside me. I usually incorporate it into my own review comments, but this time I thought sharing it "as is" might be beneficial.

---
# Review from Claude:
---

## 🛑 Blockers - Must Address Before Merge

### 1. Memory Leak Risk - ViewModel receiving Context

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsViewModel.kt:30`

**Current Code:**
```kotlin
fun init(context: Context) {
    try {
        val allLogs = AppLog.toHtmlList(context)
        _logDays.value = parseLogsByDay(allLogs)
```

**Issue:** ViewModels should not receive `Context` as a parameter. This creates a direct dependency on Android framework classes and can lead to memory leaks if the context is retained.

**Recommended Fix:**
Inject an application-scoped context via Dagger Hilt:
```kotlin
@HiltViewModel
class LogsViewModel @Inject constructor(
    private val appLogWrapper: AppLogWrapper,
    @ApplicationContext private val appContext: Context
) : ViewModel() {

    fun init() {
        try {
            val allLogs = AppLog.toHtmlList(appContext)
            _logDays.value = parseLogsByDay(allLogs)
```

Then in the Activity:
```kotlin
viewModel.init()  // No context parameter needed
```

---

### 2. ANR Risk - Heavy Operation on Main Thread

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsViewModel.kt:32`

**Current Code:**
```kotlin
fun init(context: Context) {
    try {
        val allLogs = AppLog.toHtmlList(context)  // Potentially expensive!
        _logDays.value = parseLogsByDay(allLogs)
```

**Issue:** `AppLog.toHtmlList()` could be expensive with many logs, and it's currently called on the main thread from `onCreate()`. This could cause Application Not Responding (ANR) errors with a large log file.

**Recommended Fix:**
Move to background thread with proper coroutine scope:
```kotlin
fun init(context: Context) {
    viewModelScope.launch(Dispatchers.IO) {
        try {
            val allLogs = AppLog.toHtmlList(context)
            val parsedLogs = parseLogsByDay(allLogs)
            withContext(Dispatchers.Main) {
                _logDays.value = parsedLogs
            }
        } catch (throwable: Throwable) {
            withContext(Dispatchers.Main) {
                _errorMessage.value = ErrorType.GENERAL
            }
            appLogWrapper.e(AppLog.T.SUPPORT, "Error parsing logs: ${throwable.stackTraceToString()}")
        }
    }
}
```

Consider also adding a loading state to show progress indicator while logs are loading.

---

### 3. Null Safety - Missing Navigation Guard

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsActivity.kt:96-103`

**Current Code:**
```kotlin
composable(route = LogsScreen.Detail.name) {
    val selectedLogDay by viewModel.selectedLogDay.collectAsState()
    selectedLogDay?.let { logDay ->
        LogDetailScreen(
            logDay = logDay,
            onBackClick = { navController.navigateUp() }
        )
    }
}
```

**Issue:** If `selectedLogDay` is null (e.g., due to process death/recreation, deep link, or race condition), the detail screen will be blank with no way to navigate back except the system back button.

**Recommended Fix:**
Add null handling to navigate back automatically:
```kotlin
composable(route = LogsScreen.Detail.name) {
    val selectedLogDay by viewModel.selectedLogDay.collectAsState()
    selectedLogDay?.let { logDay ->
        LogDetailScreen(
            logDay = logDay,
            onBackClick = { navController.navigateUp() }
        )
    } ?: run {
        LaunchedEffect(Unit) {
            navController.navigateUp()
        }
    }
}
```

Or consider passing the log day data through navigation arguments instead of relying on ViewModel state.

---

## ⚠️ Warnings - Should Address

### 4. Unused Property with Incorrect Scope

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsActivity.kt:34`

**Current Code:**
```kotlin
private lateinit var navController: NavHostController
```

**Issue:** `navController` is initialized inside `NavigableContent()` Composable and never accessed outside. Declaring it as a lateinit property is unnecessary and could cause issues if accessed before composition.

**Recommended Fix:**
Remove the property declaration entirely. The `navController` from `rememberNavController()` is already captured by the composable's closure:
```kotlin
@Composable
private fun NavigableContent() {
    val navController = rememberNavController()  // Use local variable only
    // ... rest of code
}
```

Also remove the unused `composeView` property (line 33).

---

### 5. Missing Lifecycle Awareness for Flow Collection

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsActivity.kt:54-65`

**Current Code:**
```kotlin
lifecycleScope.launch {
    viewModel.errorMessage.collect { errorType ->
        val errorMessage = when (errorType) {
            LogsViewModel.ErrorType.GENERAL -> getString(R.string.logs_screen_general_error)
            null -> null
        }
        errorMessage?.let {
            ToastUtils.showToast(this@LogsActivity, it, ToastUtils.Duration.LONG, Gravity.CENTER)
            viewModel.clearError()
        }
    }
}
```

**Issue:** Flow is collected without lifecycle awareness. Toasts could potentially show when the activity is in the background or destroyed.

**Recommended Fix:**
Use `repeatOnLifecycle` as done in `SupportActivity`:
```kotlin
lifecycleScope.launch {
    repeatOnLifecycle(Lifecycle.State.STARTED) {
        viewModel.errorMessage.collect { errorType ->
            val errorMessage = when (errorType) {
                LogsViewModel.ErrorType.GENERAL -> getString(R.string.logs_screen_general_error)
                null -> null
            }
            errorMessage?.let {
                ToastUtils.showToast(this@LogsActivity, it, ToastUtils.Duration.LONG, Gravity.CENTER)
                viewModel.clearError()
            }
        }
    }
}
```

---

### 6. Date Sorting is Semantically Incorrect

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsViewModel.kt:64`

**Current Code:**
```kotlin
}.sortedByDescending { it.date } // Most recent first
```

**Issue:** Sorting by string comparison on `"Oct-16"` format works accidentally within the same month but breaks across month boundaries. For example, `"Nov-01"` < `"Oct-31"` alphabetically, which would show October dates before November dates.

**Recommended Fix:**
Option 1: Parse dates for proper chronological sorting:
```kotlin
}.sortedWith(compareByDescending { logDay ->
    try {
        SimpleDateFormat("MMM-dd", Locale.getDefault()).parse(logDay.date)
    } catch (e: Exception) {
        null
    }
})
```

Option 2: Change the date format to include year and use ISO format (e.g., `"2025-10-16"`) which sorts correctly as strings.

---

## 💡 Ideas - Nice to Have

### 7. Extract URL Constant

**Location:** `WordPress/src/main/java/org/wordpress/android/support/main/ui/SupportActivity.kt:97`

**Current Code:**
```kotlin
private fun navigateToHelpCenter() {
    val intent = Intent(Intent.ACTION_VIEW, "https://wordpress.com/support/".toUri()).apply {
```

**Suggestion:**
Extract to a constant for better maintainability:
```kotlin
companion object {
    private const val HELP_CENTER_URL = "https://wordpress.com/support/"

    @JvmStatic
    fun createIntent(context: Context): Intent = Intent(context, SupportActivity::class.java)
}

private fun navigateToHelpCenter() {
    val intent = Intent(Intent.ACTION_VIEW, HELP_CENTER_URL.toUri()).apply {
```

---

### 8. Add More Granular Error Types

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsViewModel.kt:88`

**Current Code:**
```kotlin
enum class ErrorType { GENERAL }
```

**Suggestion:**
Add more specific error types for better user feedback:
```kotlin
enum class ErrorType {
    GENERAL,
    NO_LOGS_FOUND,
    PARSE_ERROR,
    IO_ERROR
}
```

---

### 9. Privacy Consideration for Log Sharing

**Location:** `WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsActivity.kt:109-124`

**Current Code:**
```kotlin
private fun shareAppLog() {
    val intent = Intent(Intent.ACTION_SEND).apply {
        type = "text/plain"
        putExtra(Intent.EXTRA_TEXT, AppLog.toPlainText(this@LogsActivity))
```

**Suggestion:**
Verify that `AppLog.toPlainText()` properly filters sensitive information such as:
- Passwords and authentication tokens
- API keys and secrets
- Personally Identifiable Information (PII)
- Email addresses and usernames
- Site URLs with authentication parameters

If not, consider adding a sanitization layer before sharing.

---

### 10. Add Analytics Tracking

**Suggestion:**
Consider tracking these events for product insights:
- When users view the Application Logs list screen
- When users open a specific log day detail
- When users share logs (helpful for support metrics)

Example:
```kotlin
composable(route = LogsScreen.List.name) {
    LaunchedEffect(Unit) {
        AnalyticsTracker.track(Stat.SUPPORT_APPLICATION_LOGS_VIEWED)
    }
    // ... rest of code
}
```

---

## ⛏ Nitpicks - Minor Issues

### 11. Test Fragility - Reflection Usage

**Location:** `WordPress/src/test/java/org/wordpress/android/support/logs/ui/LogsViewModelTest.kt:105-119, 293-310`

**Observation:**
Tests use reflection to access private methods and fields. While this provides comprehensive coverage, these tests are fragile and will break if implementation details change (method names, signatures).

**Note:** I understand the limitation here given that `AppLog.toHtmlList()` is a static method that's difficult to mock. This is acceptable for now, but consider refactoring to use dependency injection for log retrieval in the future.

---

### 12. Comment Clarity

**Location:** `WordPress/src/main/java/org/wordpress/android/support/main/ui/SupportActivity.kt:99`

**Current Code:**
```kotlin
setPackage(null) // Ensure it doesn't match internal activities
```

**Suggestion:**
Make the comment more descriptive:
```kotlin
setPackage(null) // Allow system to choose browser/handler instead of matching internal activities
```

---

## ❤️ Praise - Great Work!

1. ✅ **Clean MVVM architecture** with proper separation of concerns between Activity, ViewModel, and Composables
2. ✅ **Excellent Compose previews** for light mode, dark mode, and empty states - very helpful for development
3. ✅ **Good error handling** with try-catch blocks and proper error logging
4. ✅ **Proper navigation pattern** using sealed class for navigation events in `SupportViewModel`
5. ✅ **Comprehensive test coverage** with 20+ unit tests covering edge cases
6. ✅ **Good UX considerations** with empty states, conditional FAB visibility, and proper spacing
7. ✅ **Proper use of Jetpack Compose Navigation** with enum-based routes
8. ✅ **Analytics tracking** already added for Help Center view
9. ✅ **Monospace font** for log entries makes them easy to read
10. ✅ **Follows project patterns** by reusing existing components like `MainTopAppBar` and `AppThemeM3`

---

## Summary

This is solid work with good architecture and UX! Please address the **3 blockers** before merging:

1. **Context injection in ViewModel** (memory leak risk)
2. **Background threading for log parsing** (ANR risk)
3. **Null safety guard in navigation** (UX issue)

The warnings should ideally be addressed as well, but could be handled in a follow-up PR if time is constrained. The ideas and nitpicks are optional improvements.

**Recommendation:** Request changes for the blockers, then approve once addressed.

**Review by jkmassel — CHANGES_REQUESTED** [review]
A few suggestions:

1. We probably want to be able to share a single day's logs – I don't know if it's useful to share all of them at once? This could be a huge file with 6 days of unnecessary data to sort through.
2. I don't think we need the number of log entries in the detail view – it's helpful in the list view to give an idea of how much data they'll see by tapping.
3. Each log entry isn't really its own entity – the day is the entity as far as the user is concerned. Presentation-wise, that means we should treat all of the log entries for a given day as a single text file instead of drawing each as its own row. Technically-speaking, these files can be very long so for performance reasons it's probably a great idea to use the `LazyColumn`-based layout, but we don't need padding or styling around each line.

**Comment by adalpari** [comment]
> Testing worked as expected. As part of my PR review flow, I always ask Claude to review it alongside me. I usually incorporate it into my own review comments, but this time I thought sharing it "as is" might be beneficial.

Thank you for these great catches!
I've made some changes to fix the blockers and most of the warnings [here](https://github.com/wordpress-mobile/WordPress-Android/pull/22290/commits/70a00f6939b2d19628f7d10ed6d3e5c415d3eb57)

**Comment by adalpari** [comment]
> A few suggestions:
> 
> 1. We probably want to be able to share a single day's logs – I don't know if it's useful to share all of them at once? This could be a huge file with 6 days of unnecessary data to sort through.
> 2. I don't think we need the number of log entries in the detail view – it's helpful in the list view to give an idea of how much data they'll see by tapping.
> 3. Each log entry isn't really its own entity – the day is the entity as far as the user is concerned. Presentation-wise, that means we should treat all of the log entries for a given day as a single text file instead of drawing each as its own row. Technically-speaking, these files can be very long so for performance reasons it's probably a great idea to use the `LazyColumn`-based layout, but we don't need padding or styling around each line.

I do agree! 

1. It makes sense to have the share button per day. It seems to be up to 5 log days on Android `private const val MAX_LOG_COUNT = 5`. We are actually relying on `AppLog`  for that. Is the same on iOS? For instance, I don't see a method to clear logs, so we cannot add that button on Android.
2. I don't have a string opinion here. So, done!
3. It was just to give it some style. I've made it more plain now:

Screenshot 2025-10-17 at 10 32 44

Changes: https://github.com/wordpress-mobile/WordPress-Android/pull/22290/commits/2648ba7215f1e4d211b0e9aa82ad908d0023226e

**Review by jkmassel — APPROVED** [review]
Looks better now – we'll polish further later. Approving so we can carry on with other stuff.

File: WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    val actionEvents: SharedFlow<ActionEvent> = _actionEvents.asSharedFlow()
+
+    @Suppress("TooGenericExceptionCaught")
+    fun init() {
+        viewModelScope.launch(ioDispatcher) {
+            try {
+                val allLogs = AppLog.toHtmlList(appContext)
+                _logDays.value = parseLogsByDay(allLogs)
+            } catch (throwable: Throwable) {
+                // If there's any error parsing the logs, better not to crash the app
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/support/logs/ui/LogsViewModel.kt:61]: It might be worth tracking this.

**Review by oguzkocer — APPROVED** [review]
Looks great!
